src/bpftrace.cpp

```diff
@@ -120,29 +128,79 @@ int BPFtrace::add_probe(ast::Probe &p)
     }
     else
     {
-      attach_funcs.push_back(attach_point->func);
+      if (probetype(attach_point->provider) == ProbeType::usdt && !attach_point->ns.empty())
+        attach_funcs.push_back(attach_point->ns + ":" + attach_point->func);
+      else
+        attach_funcs.push_back(attach_point->func);
     }
 
-    for (auto func : attach_funcs)
+    for (auto func_ : attach_funcs)
```

Comment:
Similar story here, where we need the full_func_id to get the probe index, and want just func_id (FNAME) in other places.